apps/frontend/src/pages/access-management/libs/components/group-users-table/group-users-table.tsx

```diff
@@ -0,0 +1,100 @@
+import { type FieldErrors, type UseFormSetValue } from "react-hook-form";
+
+import {
+	Loader,
+	Table,
+	TablePagination,
+} from "~/libs/components/components.js";
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useEffect,
+	usePagination,
+} from "~/libs/hooks/hooks.js";
+import {
+	actions as groupActions,
+	type GroupCreateRequestDto,
+} from "~/modules/groups/groups.js";
+
+import { getUserColumns, getUserRows } from "../../helpers/helpers.js";
+import { type UserRow } from "../../types/types.js";
+import styles from "./styles.module.css";
+
+const getRowId = (row: UserRow): number => row.id;
+
+type Properties = {
+	errors: FieldErrors<GroupCreateRequestDto>;
+	onToggle: (id: number) => void;
+	selectedUserIds: number[];
+	setValue: UseFormSetValue<GroupCreateRequestDto>;
+};
+
+const GroupUsersTable = ({
+	errors,
+	onToggle,
+	selectedUserIds,
+	setValue,
+}: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+
+	const { users, usersDataStatus, usersTotalCount } = useAppSelector(
+		({ groups }) => groups,
+	);
+
+	const { onPageChange, onPageSizeChange, page, pageSize } = usePagination({
+		queryParameterPrefix: "group-user",
+		totalItemsCount: usersTotalCount,
+	});
+
+	const userColumns = getUserColumns();
+	const userData: UserRow[] = getUserRows(users);
+
+	useEffect(() => {
+		void dispatch(groupActions.loadUsers({ page, pageSize }));
+	}, [dispatch, page, pageSize]);
+
+	useEffect(() => {
+		setValue("userIds", selectedUserIds);
+	}, [selectedUserIds, setValue]);
+
+	const error = errors["userIds"]?.message;
+	const hasError = Boolean(error);
+	const hasSelectedUserIds = selectedUserIds.length === EMPTY_LENGTH;
```

Comment:
do we need this additional check at all? I think the error above should fit all the cases
```suggestion
```